element-template-generator/core/src/main/java/io/camunda/connector/generator/java/util/ReflectionUtil.java

```diff
@@ -53,4 +55,32 @@ public static List<Field> getAllFields(List<Field> fields, Class<?> type) {
     }
     return fields;
   }
+
+  public record MethodWithAnnotation<A extends Annotation>(
+      Method method, List<Parameter> parameters, A annotation) {}
```

Comment:
Do we need these values if they can be accessed via `method::getParameters` and `method::getAnnotation`?

Reply:
A method can have several annotations so its easier to have the non-nullable value as part of the result. For parameters yes, we could remove them